File: medium/49_GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) 
    {
        HashMap<String, List<String>> map = new HashMap<>();
        for(String word : strs)
        {
            //if the key doesn't exist, we add it and intialize a new list for the values
            if(!map.containsKey(sortedWord(word)))
                map.put(sortedWord(word), new ArrayList<>());
            
            //add the word to its corresponding key 
            map.get(sortedWord(word)).add(word);
        }
        return new ArrayList<>(map.values());
    }
    public String sortedWord(String s)
    {
        char[] temp = s.toCharArray();
        Arrays.sort(temp);
        String ans = "";
        for(int i = 0; i < temp.length; i++)
            ans += temp[i] + "";
        return ans;
    }
}

File: medium/49_GroupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
       
       #create a dictionary (a hashmap) of each key corresponding to a list of values
       map = collections.defaultdict(list)

       #''.join(sorted(s) standardizes each string to see if it is an anagram
       #we add the word (a value) to its corresponding key in the hashmap
       for s in strs:
            map[''.join(sorted(s))].append(s)
    
       return map.values()
       
